src/provinces/province.cpp

```diff
@@ -863,6 +863,19 @@ void change_province_owner(sys::state& state, dcon::province_id id, dcon::nation
 		}
 	}
 	for(auto r : regs) {
+		auto army = state.world.regiment_get_army_from_army_membership(r);
+		if(auto b = state.world.army_get_battle_from_army_battle_participation(army); b) {
+			for(auto& e : state.world.land_battle_get_attacker_back_line(b))
+				e = (e == r) ? dcon::regiment_id{} : e;
+			for(auto& e : state.world.land_battle_get_attacker_front_line(b))
+				e = (e == r) ? dcon::regiment_id{} : e;
+			for(auto& e : state.world.land_battle_get_defender_back_line(b))
+				e = (e == r) ? dcon::regiment_id{} : e;
+			for(auto& e : state.world.land_battle_get_defender_front_line(b))
+				e = (e == r) ? dcon::regiment_id{} : e;
+			for(auto& e : state.world.land_battle_get_reserves(b))
+				e.regiment = (e.regiment == r) ? dcon::regiment_id{} : e.regiment;
```

Comment:
you should probably just remove it from the list of reserves completely (unless you are sure that the logic for pulling them in checks for invalid ids) -- iterate from the last index in the reserves down and do the swap and pop idiom to remove